### PR TITLE
Fix some accidental globals and make a few notes for minor changes.

### PR DESCRIPTION
**Pull request by prestonhansen**
@zwnicholas  we can discuss and modify on this PR and merge once we reach an agreement on changes.

### CODE REVIEW

File: server.js [hunk 1]
```diff
@@ -10,10 +10,10 @@ var PATH = 'http://api.openweathermap.org/data/2.5/weather';
          if (request.method === 'GET') {
             
             // parsedClientReq is the parsed curl request
-            var parsedClientReq = url.parse(request.url, parseQueryString = true);
+            const parsedClientReq = url.parse(request.url, parseQueryString = true);
```
Comment (prestonhansen) [inline server.js]: this shouldnt change, so we should make it const, right?

File: server.js [hunk 2]
```diff
@@ -10,10 +10,10 @@ var PATH = 'http://api.openweathermap.org/data/2.5/weather';
...
             var apiQuery = url.parse(PATH, parseQueryString = true);
             // next three lines need to be changed to handle more robust generalized validateReq
-            var validatedQueryObj = validateReq(parsedClientReq.query);
+            const validatedQueryObj = validateReq(parsedClientReq.query);
```
Comment (prestonhansen) [inline server.js]: same here as above

File: server.js [hunk 3]
```diff
@@ -38,16 +38,16 @@ var PATH = 'http://api.openweathermap.org/data/2.5/weather';
                   }).on('end', function() {
                      resbody = Buffer.concat(resbody);
                      //console.log(`Can we print out a raw JSON property? : ${resbody.id}`);// no we can't. It's undefined.
-                     var jsn = JSON.parse(resbody);
+                     const jsn = JSON.parse(resbody);
```
Comment (prestonhansen) [inline server.js]: one more

File: server.js [hunk 4]
```diff
@@ -38,16 +38,16 @@ var PATH = 'http://api.openweathermap.org/data/2.5/weather';
...
                      console.log(`The weather in ${jsn.name} is ${jsn.weather[0].description}. It\'s 
                            currently ${jsn.main['temp']} degrees K.`);
                         //console.log(typeof jsn.id); // it's a number! That's why we need to toString() it below
 
                      response.writeHead(200, {"Content-Type": "text/plain"})
                      // you can only write back strings or buffers!!!;
-                     response.end(`${jsn.name.toString()}, ${jsn.sys.country.toString()}:
-                            ${jsn.weather[0].description.toString()} and ${jsn.main.temp.toString()} degrees K`);
+                      response.end(`${jsn.name.toString()}, ${jsn.sys.country.toString()}:
+                            ${jsn.weather[0].description} and ${jsn.main.temp.toString()} degrees K`);
```
Comment (prestonhansen) [inline server.js]: `jsn.weather[0].description` should already be a string as opposed to `jsn.main.temp`, so we shouldn't need to convert it with `.toString()`, right?

File: server.js [hunk 5]
```diff
@@ -71,8 +71,8 @@ var PATH = 'http://api.openweathermap.org/data/2.5/weather';
       // AND A ERROR OR SUCCESS FLAG
       // mainObj.cities will be a list of city objects name and ID keys 
       var mainObj = {};
-      len1Re = /[A-Za-z\u00A0-\uD7FF\uF900-\uFDCF\uFDF0-\uFFEF]/;
-      trimmed = clientReq['q'].split(',').map(function (element) {
+      const len1Re = /[A-Za-z\u00A0-\uD7FF\uF900-\uFDCF\uFDF0-\uFFEF]/;
```
Comment (prestonhansen) [inline server.js]: this variable (and trimmed below) were declared without a keyword so they are globals. probably don't want that. 

made the regex const -- it shouldn't change. trimmed might be able to be const as well, but that depends on how you use it.

File: server.js [hunk 6]
```diff
@@ -93,20 +93,21 @@ var PATH = 'http://api.openweathermap.org/data/2.5/weather';
...
-            console.log(`it exceedingly unlikely that we should be here`);
+            console.log(`it is exceedingly unlikely that we should be here`);
             mainObj.flag = 'ERROR: UNLIKELY';
             return mainObj;
          }
       }
       else { // assume client is searching by city
-        city = trimmed[0];
-        country = trimmed[1];
+        const city = trimmed[0];
```
Comment (prestonhansen) [inline server.js]: same deal as the others.

File: server.js [hunk 7]
```diff
@@ -93,20 +93,21 @@ var PATH = 'http://api.openweathermap.org/data/2.5/weather';
          }
          else if (trimmed.match(len1Re)) {
             mainObj.zip = -1;
+	    //pretty sure this wont work since city isn't defined yet...
```
Comment (phansen01) [inline server.js]: `cityDB` is a global from above, but `city` is not defined yet unless i'm missing something (not getting any hits on it until below this conditional branch). Not sure exactly what the intention is and/or how to correctly change this so I will leave that up in the air.

**Comment by prestonhansen** [comment]
@zwnicholas  i've commented on all of the diffs in this commit, let me know what you think and we can make final changes as necessary.

File: server.js [hunk 8]
```diff
@@ -93,20 +93,21 @@ var PATH = 'http://api.openweathermap.org/data/2.5/weather';
...
             mainObj.cities = findID(cityDB, city);
```
Comment (prestonhansen) [inline server.js:105]: cityDB is a global from above, but city is not defined yet unless i'm missing something (not getting any hits on it until below this conditional branch). Not sure exactly what the intention is and/or how to correctly change this so I will leave that up in the air.

**Pull request merged**